### PR TITLE
Dependency fix

### PR DESCRIPTION
**Pull request by Pzixel**
I tried to dogfood with cv-rs and solverd several problems. However, it currently doesn't work as it fails on linking phase with `undefined reference to `cv::img_hash::PHash::create()'`. 

We may switch to bindgen, but we still need higher level abstraction, so it worth to continue adding them.

Looking forward if you could help with the issue.

### CODE REVIEW

**Comment by nebgnahz** [comment]
On Linux (Travis), we will need to manual add `println!("cargo:rustc-link-lib=opencv_img_hash");` around [line 65 of `build.rs`](https://github.com/Pzixel/cv-rs/blob/1216c32bc6062d89c9545a30282b6d6e9d5b72a8/build.rs#L65).

On Windows (Circle), I am not sure what causes the issue given things are compiled into `opencv_world`. But I suspect a similar issue with linking.

File: native/phash.cc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#ifndef PHASH_H_
+#define PHASH_H_
```
Comment (nebgnahz) [inline native/phash.cc]: The guards should not appear in `.cc` files.

Comment (Pzixel) [reply native/phash.cc]: We had it in `utils.cc`. However, you're right.

**Comment by Pzixel** [comment]
Well, currently I have two problems:

1. rustfmt shows the difference although I just updated the chain and used `cargo fmt`.
2. it does compile and runs on linux, but it doesn't on windows. No clue why it's happening.

**Comment by Pzixel** [comment]
One single screenshot to answer 😄 them all

image

File: build.rs [hunk 2]
```diff
@@ -1,77 +1,124 @@
...
+            println!("cargo:rustc-link-lib=opencv_cudaobjdetect");
+        }
     }
 }
 
+#[cfg(windows)]
+use windows::*;
+
+#[cfg(unix)]
+use unix::*;
```
Comment (nebgnahz) [inline build.rs]: nice!

**Comment by nebgnahz** [comment]
Cool! LGTM.

**Pull request merged**